swagger.json

```diff
@@ -5548,6 +5773,59 @@
             "in": "query",
             "type": "string"
         },
+        "addressAddParam": {
+            "name": "addresses",
+            "description": "Addresses/Geofences to add. Geofences can be circular or a polygon. \nFor each address, only one of 'circle' or 'polygon' should be provided. If both are provided, the geofence will be saved as a polygon.",
+            "required": true,
+            "in": "body",
+            "schema": {
+                "type": "object",
+                "required": [
+                    "name",
+                    "formattedAddress",
+                    "geofence"
+                ],
+                "properties": {
+                    "name": {
+                        "type": "string",
+                        "description": "The name of this address/geofence",
+                        "example": "Samsara HQ"
+                    },
+                    "formattedAddress": {
+                        "type": "string",
+                        "description": "The full address associated with this address/geofence, as it might be recognized by maps.google.com",
+                        "example": "350 Rhode Island St, San Francisco, CA"
+                    },
+                    "geofence": {
+                        "$ref": "#/definitions/AddressGeofence"
+                    }
+                }
+            }
+        },
+        "addressUpdateParam": {
+            "name": "address",
+            "description": "Update parts of an address's value. If the geofence 'circle' or 'polygon' key is specified, the update will change the type of geofence accordingly.",
+            "required": true,
+            "in": "body",
+            "schema": {
+                "type": "object",
```

Comment:
Nit: This shares the same schema as above, with the exception of the requires parameters.

Reply:
The schema of the update vs the add seemed substantially different in that effectively every field is optional in the update. Wasn’t sure if there was a cleaner way to handle that in swagger

Reply:
I believe you can reference definitions and still specify a different set of `required` attributes. In any case, this is fine as it is, just means that we'll need to remember to update two call sites if we change the request/response structure.

Reply:
I tried that earlier, but it didn’t seem to a work. It could also be that I didn’t format my fields correctly